### PR TITLE
Add skip setup option in design picker

### LINKED ISSUE #97853: Task: Allow skipping the Design Picker
### Details

p1735598906999299/1735597737.159869-slack-C085HCWCEDN

### Checklist

- [ ]
- [ ]

### Related

_No response_

### PR DESCRIPTION
**Pull request by madhusudhand**
Closes https://github.com/Automattic/dotcom-forge/issues/10094. Closes https://github.com/Automattic/wp-calypso/issues/97853.

## Proposed Changes

* Add skip option in design picker when goal centric feature flag is enabled.

image

image

## Testing Instructions

* Navigate to design picker
* It should have `skip setup` button on top right as suggested by the design.
* Enter other flows such as `/setup/update-design?siteSlug=%s` and verify the skip button isn't there

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by fditrapani** [comment]
Thanks for putting this together @madhusudhand. Can we hold off on merging it until we implement the skip setup button across the entire flow?

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: I think we should use `isGoalCentricFeature` or it should be `isGoalsHoldout`, right?

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:917]: It would be better to keep `Skip` to align with other steps. What do you think? cc @fditrapani

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (madhusudhand) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: > from [this comment](https://github.com/Automattic/dotcom-forge/issues/10094#issuecomment-2548590658), This should only be for the hold out test! Thanks for confirming.

I negated to `hide the skip` when `isGoalsHoldout` is Disabled(false) - previous behaviour, and `show skip` when it is Enabled. @fditrapani could you please confirm.

Comment (arthur791004) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: > https://github.com/Automattic/dotcom-forge/issues/10094#issuecomment-2547383295

Referring to [previous one](https://github.com/Automattic/dotcom-forge/issues/10094#issuecomment-2547383295), I assume it means we have to keep it hidden for people under the holdout test (the value of `isGoalsHoldout` is true or the value of `isGoalCentricFeature` is true). I was confused about the similar description before 🙈

Comment (madhusudhand) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: Thanks you for clarifying 🙇 Updated the condition to hide skip when `isGoalsHoldout` is `true`

**Review by arthur791004 — APPROVED** [review]
LGTM 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (fditrapani) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:917]: Thanks for bringing this up @arthur791004. There have been discussions to have a single persistent `Skip setup` link on all the onboarding steps that behaves the same across the board. We'll tackle the other steps separately and can then merge this in together with the others when they're ready.

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (fditrapani) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: Thanks for checking in. We only want to show this link within the new onboarding flow.

**Pull request merged**
